subiquitycore/utils.py

```diff
@@ -234,21 +235,41 @@ def log_process_streams(
     log.log(level, "--------------------------------------------------")
 
 
-# FIXME: replace with passlib and update package deps
+def _generate_salt() -> str:
```

Comment:
Could we do the following instead of manually generating a salt and slicing?

```python
handler.using(salt_chars="abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ0123456789./")
```

I believe we should be able to mock.patch https://passlib.readthedocs.io/en/stable/lib/passlib.utils.handlers.html?highlight=_generate_salt#passlib.utils.handlers.HasSalt._generate_salt

Reply:
So double checking the docs (e.g. for [sha512](https://passlib.readthedocs.io/en/stable/lib/passlib.hash.sha512_crypt.html)), if unspecified a salt will be generated automatically and the default salt character set is  `[./0-9A-Za-z]` for all of the algorithms. So yeah we can remove the manual salt generation entirely now, and `@patch("passlib.utils.handlers.HasSalt._generate_salt")` works great.

Reply:
Excellent